CMake/FindJulia.cmake

```
@@ -22,6 +22,17 @@ if (JULIA_EXECUTABLE)
   endif ()
 endif ()
 
+# If force Julia binding fails then it must not effect Go bindings
+# in next "cmake ../".
+if (NOT JULIA_EXECUTABLE AND FORCE_BUILD_JULIA_BINDINGS)
+  unset(BUILD_GO_BINDINGS CACHE)
+endif()
+
+if (NOT JULIA_EXECUTABLE)
+  unset(BUILD_JULIA_BINDINGS CACHE)
+  unset(FORCE_BUILD_JULIA_BINDINGS CACHE)
+endif()
```

```suggestion
if (NOT JULIA_EXECUTABLE)
  unset(BUILD_JULIA_BINDINGS CACHE)
endif()
```

Correct me if I'm wrong, but I think we should be able to simplify to just this.

---

But after this --- if we pass `-DBUILD_JULIA_BINDINGS=ON` and if it fails and then we just pass a simple `cmake ../` It will fail with a error, if required dependencies of go bindings are not install. May because it is not reaching FindGo.cmake or Cmakelists(now) of go bindings to unset the cache.

---

Updated this part. Moved this too in `CMakeLists.txt` of Julia bindings.